service/header/header.go

```diff
@@ -59,20 +59,17 @@ func (eh *ExtendedHeader) ValidateBasic() error {
 		return err
 	}
 
-	if eh.Commit.Height != eh.Height {
-		return fmt.Errorf("header and commit height mismatch: %d vs %d", eh.Height, eh.Commit.Height)
-	}
-	if hhash, chash := eh.Hash(), eh.Commit.BlockID.Hash; !bytes.Equal(hhash, chash) {
-		return fmt.Errorf("commit signs block %X, header is block %X", chash, hhash)
-	}
```

Comment:
If you look closer, this line is nonsense because it checks equality between the same value(eh.Hash() reads from https://github.com/Wondertan)